Plugins/OpenDroneID2/NMEA_GPS_Connection.cs

```diff
@@ -0,0 +1,402 @@
+﻿using MissionPlanner.Utilities;
+using MissionPlanner.Comms;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Globalization;
+using System.Linq;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using System.Diagnostics;
+using LibVLC.NET;
+using MissionPlanner.Controls;
+
+namespace MissionPlanner
+{
+    public partial class NMEA_GPS_Connection : UserControl
+    {
+
+        static TcpListener listener;
+        static ICommsSerial comPort = null;
+        //static internal PointLatLngAlt lastgotolocation = new PointLatLngAlt(0, 0, 0, "Goto last");
+        //static internal PointLatLngAlt gotolocation = new PointLatLngAlt(0, 0, 0, "Goto");
+
+
+        public double Lat, Lng;
+        public double Alt, Alt_WGS84;
+        public float hdop;
+        int sats;
+        int fix_type;
+        public Stopwatch last_gps_msg = new Stopwatch();
+        private bool portsAreLoaded = false;
+        static NMEA_GPS_Connection Instance;
+
+        private NMEA_Viewer _nmeaViewer;
+
+        public NMEA_GPS_Connection()
+        {
+            InitializeComponent();
+            Instance = this;
+
+            try
+            {
+                init_com_port_list();
+            }
+            catch
+            {
+                Console.WriteLine("Couldn't Init Open DID Form.");
+            }
+            timer2.Start();
+        }
+
+        private void timer1_Tick(object sender, EventArgs e)
```

Comment:
can we background thread this operation?
the reason is if for whatever reason we dont read a full line, or there is some serial io issue, the UI will hang for the timeout period.

Reply:
yes, that would be a fairly minor change. I can have it done before your tomorrow morning.